Sources/VectorExtor/CGPath+Extensions.swift

```diff
@@ -3,275 +3,152 @@ import CoreGraphics
 
 @available(OSX 10.13, iOS 11.0, tvOS 11.0, watchOS 4.0, *)
 public extension CGPath {
-	enum PathElement: Equatable {
-		case moveTo(point: CGPoint)
-		case addLineTo(point: CGPoint)
-		case addQuadCurveTo(point: CGPoint, controlPoint: CGPoint)
-		case addCurveTo(point: CGPoint, controlPoint1: CGPoint, controlPoint2: CGPoint)
-		case close
-	}
 
-	class PathSection {
-		public let element: PathElement
-		var next: PathSection?
-//		var previous: PathSection?
-		weak var previous: PathSection?
+	struct Segment: Hashable, Codable, Sendable {
+		public enum Curve: CurveProtocol, Hashable, Codable, Sendable {
+			case moveTo(MoveCurve)
+			case addLineTo(LineCurve)
+			case addQuadCurveTo(QuadCurve)
+			case addCurveTo(CubicCurve)
+			case close(CloseCurve)
+
+			public var length: Double {
+				switch self {
+				case
+					.addLineTo(let curve as CurveProtocol),
+					.addQuadCurveTo(let curve as CurveProtocol),
+					.addCurveTo(let curve as CurveProtocol),
+					.close(let curve as CurveProtocol),
+					.moveTo(let curve as CurveProtocol):
+					curve.length
+				}
+			}
 
-		var endPoint: CGPoint? {
-			switch element {
-			case .addLineTo(point: let point):
-				return point
-			case .addCurveTo(point: let point, controlPoint1: _, controlPoint2: _):
-				return point
-			case .addQuadCurveTo(point: let point, controlPoint: _):
-				return point
-			case .moveTo(point: let point):
-				return point
-			default:
-				return nil
+			public var startPoint: CGPoint? {
+				switch self {
+				case .moveTo(let curve):
+					curve.startPoint
```

Comment:
I've seen this syntax used where the return keyword is omitted.  To my mind, this reduces readability.  The more the reader has to interpret or add in their mind as they read the code, the worse it gets.  This is I guess just a style thing, but for me it's an example of where Swift is heading down the wrong path.

Reply:
I disagree on the wrong path thing. I do that as a means of reducing code clutter. While I personally feel like it improves my readability (by having less visual clutter), I can understand the counter argument. However, as you mentioned this is a stylistic choice I made.